holoviews/plotting/bokeh/element.py

```diff
@@ -260,6 +263,28 @@ def _get_hover_data(self, data, element, empty=False):
                 data[dim] = [v for _ in range(len(list(data.values())[0]))]
 
 
+    def _merge_ranges(self, plots, xlabel, ylabel):
+        """
+        Given a list of other plots return axes that are shared
+        with another plot by matching the axes labels
+        """
+        plot_ranges = {}
+        for plot in plots:
+            if plot is None:
+                continue
+            if hasattr(plot, 'xaxis'):
+                if plot.xaxis[0].axis_label == xlabel:
+                    plot_ranges['x_range'] = plot.x_range
+                if plot.xaxis[0].axis_label == ylabel:
+                    plot_ranges['y_range'] = plot.x_range
+            if hasattr(plot, 'yaxis'):
+                if plot.yaxis[0].axis_label == ylabel:
+                    plot_ranges['y_range'] = plot.y_range
+                if plot.yaxis[0].axis_label == xlabel:
+                    plot_ranges['x_range'] = plot.y_range
+        return plot_ranges
```

Comment:
That is correct, I could have two separate loops and break when either is found first, but in practice if ``shared_axes`` are enabled all the plots will share the same axes anyway so it doesn't much matter whether I end up using the axis from the first or last plot, because they are all the same anyway.